lib/README.md

```diff
@@ -0,0 +1,148 @@
+# Gutenberg PHP
+
+## File structure
+
+Gutenberg adds features to WordPress Core using PHP hooks and filters. Some
+features, once considered stable and useful, are merged into Core during a Core
+release. Some features remain in the plugin forever or are removed.
+
+To make it easier for contribtors to know which features need to be merged to
+Core and which features can be deleted, Gutenberg uses the following file
+structure for its PHP code:
+
+- `lib/experimental` - Experimental features that exist only in the plugin. They
+  are not ready to be merged to Core.
+- `lib/stable` - Stable features that exist only in the plugin. They could one
+  day be merged to Core, but not yet.
+- `lib/compat/wordpress-X.Y` - Stable features that are intended to be merged to
```

Comment:
Not suggesting we do it right away, but do  you think it'd be helpful for folks if expanded the compat topic at some stage? For example rationale, preferred naming conventions when extending classes, things to test, and so on. 

There are also plenty of lessons to draw from PRs such as https://github.com/WordPress/gutenberg/pull/38883 and https://github.com/WordPress/gutenberg/pull/38671

Or maybe it's just too fuzzy an area to pin down. 🌪️

Reply:
Yeah I think we're still figuring a lot of that out, so it's hard to pen down. If you have any suggestions on things we can add now then please go ahead!